application-ckeditor-plugins/src/main/resources/xwiki-image/plugin.js

```diff
@@ -17,224 +17,117 @@
-              // This is an internal class that should not be visible when editing the image.
-              image.removeClass('wikimodel-freestanding');
-              // The alt attribute is auto-generated for free-standing images. We remove it in order to be consistent
-              // with the wiki syntax where it's not specified.
-              delete image.attributes.alt;
-            }
-          } else {
-            // Unexpected HTML structure inside image markers. Keep the markers.
-            return false;
-          }
-        },
-        // element: CKEDITOR.htmlParser.element
-        isMarked: function(element) {
-          return element.name === 'img' && element.attributes['data-reference'];
-        },
-        // image: CKEDITOR.htmlParser.element
-        toDataFormat: function(image) {
-          // Put back the image rendering markers.
-          var reference = CKEDITOR.tools.escapeComment(image.attributes['data-reference']);
-          var startImageComment = new CKEDITOR.htmlParser.comment('startimage:' + reference);
-          var stopImageComment = new CKEDITOR.htmlParser.comment('stopimage');
-          startImageComment.insertBefore(image);
-          stopImageComment.insertAfter(image);
-          delete image.attributes['data-reference'];
-          // Handle free-standing images.
-          var freeStanding = image.attributes['data-freestanding'] === 'true';
-          delete image.attributes['data-freestanding'];
-          // Free-standing images don't have white-space in their reference and have only the src attribute set.
-          if (freeStanding && !/\s+/.test(reference) && isFreeStandingImage(image)) {
-            image.addClass('wikimodel-freestanding');
-          }
+  function showImageWizard(editor, widget) {
+    require(['imageWizard'], function(imageWizard) {
+      imageWizard({
+        editor: editor,
+        macroData: widget.data
+      }).done(function(data) {
+        if (widget && widget.element) {
+          widget.setData(data);
+        } else {
+          var element = CKEDITOR.dom.element.createFromHtml(widget.template.output(), editor.document);
+          var wrapper = editor.widgets.wrapElement(element, widget.name);
+          var temp = new CKEDITOR.dom.documentFragment(wrapper.getDocument());
+
+          // Append wrapper to a temporary document. This will unify the environment in which #data listeners work when
+          // creating and editing widget.
+          temp.append(wrapper);
+
+          var instance = editor.widgets.initOn(element, widget, {});
+          instance.setData(data);
+          editor.widgets.initOn(element, widget, data);
+          editor.widgets.finalizeCreation(temp);
         }
       });
+    });
+  }
 
-      editor.balloonToolbars.create({
-        buttons: 'Link,Unlink,Image',
-        widgets: 'image'
-      });
+  CKEDITOR.plugins.add('xwiki-image', {
+    requires: 'xwiki-image-old', // TODO: maybe require modal to have skinx?
```

Comment:
Yes, should depend on modal too.

Reply:
Hum, actually I'm not sure how to depend from modal here.

Reply:
Answering myself, with `xwiki-dialog`.